src/sshFileSystem.ts

```diff
@@ -138,6 +143,39 @@ export class SSHFileSystem implements vscode.FileSystemProvider {
   public rename(oldUri: vscode.Uri, newUri: vscode.Uri, options: { overwrite: boolean; }): void | Promise<void> {
     return this.continuePromise(cb => this.sftp.rename(this.relative(oldUri.path), this.relative(newUri.path), cb));
   }
+
+  public async provideFileIndex(options: vscode.FileSearchOptions, token: vscode.CancellationToken): Promise<vscode.Uri[]> {
+    const name = this.config.label || this.config.name;
+    const command = `find ${this.relative(options.folder.path)} -print0`;
```

Comment:
Have you checked whether this command and syntax is available on all distros that support SFTP?